docs/cookbook/entities/first-resource.rst

```diff
@@ -46,6 +46,43 @@ As an example we will take an **Article entity**.
         }
     }
 
+
+If you don't add a form type, it uses a `default form type`_. But it is a good practice to have one.
+
+.. code-block:: php
+
+    // src/Form/Type/ArticleType.php
+
+    namespace App\Form\Type;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\Extension\Core\Type\TextType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    class ArticleType extends AbstractType
+    {
+        /**
+        * {@inheritdoc}
+        */
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            parent::buildForm($builder, $options);
+            $builder
+                ->add('title', TextType::class, [
+                    'label' => 'sylius.ui.title',
+                ]);
+        }
+
```

Comment:
I think this method is missing (and also add use App/Entity/Article).

    /**
     * {@inheritdoc}
     */
    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults([
            'data_class' => Article::class
        ]);
    }

Reply:
Is this really needed ? I have used this class without these methods. Also this was taken from https://github.com/Monofony/SymfonyStarter/blob/e93c5a820c042a9770dd7017a94547cc6a239e0b/src/Form/Type/ArticleType.php . And it worked with out any issues.

Reply:
@harikt When you need embedded form, it will break.
look at the [Symfony doc](https://symfony.com/doc/current/forms.html) on **Setting the data_class** part
It's not mandatory but source of bugs later.

Reply:
Thank you for pointing out the reason. Fixed it.